app/controllers/downloads_controller.rb

```diff
@@ -2,16 +2,27 @@
 
 # DownloadsController: for handling user downloads
 class DownloadsController < ApplicationController
-  before_action :authenticate_user!
+  include Rails.application.routes.url_helpers
+  before_action :authenticate_user!, :current_user_download
 
   def download
     redirect_to root_url unless current_user.can_export?
-    lookup = params[:lookup]
-    download = current_user.downloads.find_by(lookup: lookup)
-    if download.nil?
-      @error = true
-    else
-      send_data(Base64.decode64(download.contents), filename: download.filename) && download.delete
-    end
+    @export_url = rails_storage_proxy_path(@download.exports.first)
```

Comment:
If logged in as a different user than the user that initiated the download, then `@download` will be `nil` and L10 will cause an exception. Could we add a check on L9 that just redirects to root if `@download.nil?`

Reply:
I changed `find_by` to `find_by!` so the `before_action` will throw a not found error before we reach this.